src/esp/sensor/VisualSensor.h

```diff
@@ -81,14 +93,51 @@ class VisualSensor : public Sensor {
     return false;
   }
 
+  /**
+   * @brief Sets resolution of Sensor's sensorSpec
+   */
+  void setResolution(int height, int width);
+  void setResolution(vec2i resolution);
+
   /**
    * @brief Returns RenderCamera
    */
   virtual gfx::RenderCamera* getRenderCamera() = 0;
 
+  /**
+   * @brief Gets near plane distance.
+   */
+  float getNear() { return near_; }
+
+  /**
+   * @brief Gets far plane distance.
+   */
+  float getFar() { return far_; }
+
+  /**
+   * @brief Returns the FOV of this Sensor
+   */
+  Mn::Deg getFOV() const { return hfov_; }
+
  protected:
-  std::unique_ptr<gfx::RenderTarget> tgt_;
+  /** @brief projection parameters
+   */
+
+  /** @brief near clipping plane
+   */
+  float near_ = 0.01f;
+
+  /** @brief far clipping plane
+   */
+  float far_ = 1000.0f;
 
+  /** @brief field of view
+   */
+  Mn::Deg hfov_ = 90.0_degf;
```

Comment:
kind note: PR would be a breaking change to the habitat-lab. 
You may need another PR to fix it in the habitat-lab repository.